test/e2e/issuer_acme_dns01_test.go

```diff
@@ -0,0 +1,1174 @@
+		policy.Bindings[bindingIndex].Members = members
+
+		// Remove binding if no members left
+		if len(members) == 0 {
+			policy.Bindings = append(policy.Bindings[:bindingIndex], policy.Bindings[bindingIndex+1:]...)
+		}
+	}
+
+	// updateGCPIamPolicyBinding adds or removes an IAM policy binding for a GCP project
+	updateGCPIamPolicyBinding := func(crmService *gcpcrm.Service, projectID, role, member string, add bool) error {
+		return wait.PollUntilContextTimeout(context.TODO(), fastPollInterval, lowTimeout, true, func(context.Context) (bool, error) {
+			policy, err := crmService.Projects.GetIamPolicy(projectID, &gcpcrm.GetIamPolicyRequest{}).Do()
+			if err != nil {
+				log.Printf("Error getting IAM policy: %v", err)
+				return false, nil
+			}
+
+			if add {
+				policy.Bindings = append(policy.Bindings, &gcpcrm.Binding{
+					Role:    role,
+					Members: []string{member},
+				})
+			} else {
+				removeGCPMemberFromPolicy(policy, role, member)
+			}
+
+			_, err = crmService.Projects.SetIamPolicy(projectID, &gcpcrm.SetIamPolicyRequest{Policy: policy}).Do()
+			if err != nil {
+				log.Printf("Error setting IAM policy: %v", err)
+				return false, nil
+			}
+			log.Printf("IAM policy updated successfully")
+			return true, nil
+		})
+	}
+
+	Context("with AWS Route53", Label("Platform:AWS", "CredentialsMode:Mint"), func() {
+
+		It("should obtain a valid certificate using explicit credentials", func() {
+
+			// Get AWS credentials and region
+			accessKeyID, secretAccessKey := getAWSCredentials(ctx)
+			region := getAWSRegion(ctx)
+
+			// Copy secret to test namespace
+			secretName := "aws-secret"
+			secretKey := "aws_secret_access_key"
+			copyAWSSecretToNamespace(ctx, ns.Name, secretName, secretKey, secretAccessKey)
+
+			By("creating ACME issuer with Route53 DNS-01 solver using explicit credentials")
+			issuerName := "letsencrypt-dns01"
+			solver := acmev1.ACMEChallengeSolver{
+				DNS01: &acmev1.ACMEChallengeSolverDNS01{
+					Route53: &acmev1.ACMEIssuerDNS01ProviderRoute53{
+						Region:      string(region),
+						AccessKeyID: string(accessKeyID),
+						SecretAccessKey: certmanagermetav1.SecretKeySelector{
+							LocalObjectReference: certmanagermetav1.LocalObjectReference{
+								Name: secretName,
+							},
+							Key: secretKey,
+						},
+					},
+				},
+			}
+			issuer := createACMEIssuer(issuerName, solver)
+			_, err := certmanagerClient.CertmanagerV1().Issuers(ns.Name).Create(ctx, issuer, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred(), "failed to create issuer")
+
+			By("waiting for issuer to become ready")
+			err = waitForIssuerReadiness(ctx, issuerName, ns.Name)
+			Expect(err).NotTo(HaveOccurred(), "timeout waiting for issuer to become Ready")
+
+			// Create and verify certificate
+			certName := "letsencrypt-cert"
+			dnsName := fmt.Sprintf("adre-%s.%s", randomStr(3), appsDomain)
```

Comment:
nit: The acronyms can be expanded, like what's ADRE. Not a must, this can be derived from the `By` step, but good to have.